internal/database/lock.go

```diff
@@ -50,39 +52,44 @@ func (db *DB) MultiLock(ctx context.Context, lockIDs []string, ttl time.Duration
 	sort.Strings(lockOrder)
 
 	var expires time.Time
-	err := db.InTx(ctx, pgx.Serializable, func(tx pgx.Tx) error {
+	if err := db.InTx(ctx, pgx.ReadCommitted, func(tx pgx.Tx) error {
 		for _, lockID := range lockOrder {
-			row := tx.QueryRow(ctx, `SELECT AcquireLock($1, $2)`, lockID, int(ttl.Seconds()))
+			row := tx.QueryRow(ctx, `SELECT AcquireLock($1, $2)`, lockID, int32(ttl.Seconds()))
 			if err := row.Scan(&expires); err != nil {
-				return err
+				return fmt.Errorf("failed to scan multilock.expires: %w", err)
 			}
 			if expires.Before(thePast) {
 				return ErrAlreadyLocked
 			}
 		}
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
 		return nil, err
 	}
-	logging.FromContext(ctx).Debugf("Acquired locks %v", lockOrder)
+
+	logger.Debugw("acquired locks", "locks", lockOrder)
 	return makeMultiUnlockFn(ctx, db, lockOrder, expires), nil
 }
 
 func makeMultiUnlockFn(ctx context.Context, db *DB, lockIDs []string, expires time.Time) UnlockFn {
+	logger := logging.FromContext(ctx).Named("database.MultiUnlock")
+
 	return func() error {
-		return db.SerializableTx(ctx, func(tx pgx.Tx) error {
-			for i := len(lockIDs) - 1; i >= 0; i-- {
-				lockID := lockIDs[i]
+		return db.InTx(ctx, pgx.ReadCommitted, func(tx pgx.Tx) error {
+			for _, lockID := range lockIDs {
+				logger := logger.With("lock_id", lockID)
+
 				row := tx.QueryRow(ctx, `SELECT ReleaseLock($1, $2)`, lockID, expires)
 				var released bool
 				if err := row.Scan(&released); err != nil {
-					return err
+					return fmt.Errorf("failed to scan lock.released: %w", err)
 				}
-				if !released {
-					return fmt.Errorf("cannot delete lock %q that no longer belongs to you; it likely expired and was taken by another process", lockID)
```

Comment:
⚠️ Behavior change. It feels wrong (and makes testing _very_ difficult) to return an error if the lock was failed to release. With the new function, releasing the lock will _always_ succeed if the procedure finishes.